BrowserKit/Tests/WebEngineTests/WKEngineSessionTests.swift

```diff
@@ -285,7 +306,16 @@ final class WKEngineSessionTests: XCTestCase {
                                             contentScriptManager: contentScriptManager) else {
             return nil
         }
+
         trackForMemoryLeaks(subject, file: file, line: line)
+
         return subject
     }
+
+    // Adding a special teardown since as part of tracking memory leaks, we can't use an instance variable on the
+    // test suite. A normal instance `tearDown` is called after a `addTeardownBlock`. To ensure we still can
+    // `trackForMemoryLeaks` we need to always close any engine session that is opened, otherwise leaks happens.
+    func prepareForTearDown(_ subject: WKEngineSession) {
+        subject.close()
```

Comment:
I might be misunderstanding, but is this `close()` necessary in order to avoid a retain cycle that will leak the instance? Is this something that could potentially impact the client?

Reply:
Well, I'm still trying to figure out who will be responsible for the `close()` call, but we'll need to call this for sure when removing a webview/session/tab from the application 🤔